oioswift/common/middleware/container_hierarchy.py

```diff
@@ -249,13 +249,14 @@ def _create_placeholder(self, req, account, container, mode, path):
         # to avoid false counter
         if self.redis_keys_format == REDIS_KEYS_FORMAT_V1:
             res = self.conn.set(key, "1")
-            if not res:
-                LOG.warn("%s: failed to create key %s", self.SWIFT_SOURCE, key)
+            if not res and not self.conn.exists(key):
+                LOG.error("%s: failed to create key %s",
+                          self.SWIFT_SOURCE, key)
         else:
             res = self.conn.hset(key, path, "1")
-            if not res:
-                LOG.warn("%s: failed to create key %s %s",
-                         self.SWIFT_SOURCE, key, path)
+            if not res and not self.conn.hexists(key, path):
+                LOG.error("%s: failed to create key %s %s",
+                          self.SWIFT_SOURCE, key, path)
```

Comment:
If this is an error, why do we continue normal execution?

Reply:
Ok, the error was not here. A real error throws an exception, and thus terminates the request execution.